src/go/rpk/pkg/api/admin/api_user.go

```diff
@@ -51,6 +51,23 @@ func (a *AdminAPI) CreateUser(ctx context.Context, username, password, mechanism
 	return a.sendToLeader(ctx, http.MethodPost, usersEndpoint, u, nil)
 }
 
+// UpdateUser updates a user with the given username and password using the
+// given mechanism (SCRAM-SHA-256, SCRAM-SHA-512).
+func (a *AdminAPI) UpdateUser(ctx context.Context, username, password, mechanism string) error {
+	if username == "" {
+		return errors.New("invalid empty username")
+	}
+	if password == "" {
+		return errors.New("invalid empty password")
+	}
+	u := newUser{
+		User:      username,
+		Password:  password,
+		Algorithm: mechanism,
+	}
+	return a.sendToLeader(ctx, http.MethodPut, usersEndpoint, u, nil)
```

Comment:
I think the endpoint must have the user and the user is not needed in the body:

```
$ curl -X PUT localhost:9644/v1/security/users/foo \
  -d '{"password":"bar","algorithm":"SCRAM-SHA-256"}' 
```

So it'll be something like usersEndpoint/user

Reply:
https://github.com/redpanda-data/redpanda/blob/dev/src/v/redpanda/admin/api-doc/security.json#L39-L56

Reply:
Nice, thank you for letting me know. Would you happen to know is the post and put endpoints different?

Reply:
No idea :sweat_smile:

Reply:
Does the user object does not allow to omit? In either case, i will send the user data as part of the payload.

Reply:
What user data as a part of the payload? I don't see a reason to send the user in the PUT body, we can use an inline struct,
```
u := struct {
  Password string `json:"password"`
  Mechanism string `json:"mechanism"`
}{password, mechanism}
```
?